src/pygeoprocessing/routing/routing.pyx

```diff
@@ -2758,6 +2774,8 @@ def distance_to_channel_mfd(
                         else:
                             weight_val = (SQRT2 if i_n % 2 else 1)
 
+                        if pixel.value == distance_nodata:
+                            pixel.value = 0
                         pixel.value += flow_dir_weight * (

```

Comment:
Oh interesting. I assumed it would be the shortest distance (following the shortest path of pixels that have any MFD proportion > 0). But it's really more like the average distance that water travels from that pixel to the outlets.